planning/route_handler/src/route_handler.cpp

```diff
@@ -293,6 +293,7 @@ void RouteHandler::setLaneletsFromRouteMsg()
     return;
   }
   for (const auto & route_section : route_msg_.segments) {
+    route_lanelets_.reserve(route_lanelets_.size() + route_section.primitives.size());
```

Comment:
Is it okay to reserve `route_lanelets_` here?
(Should it be reserved before repeating (L295)?

Reply:
because we can't guarantee the size of `route_section.primitive`, so it can't be reserve before line 295.

Reply:
Okay, I see.
One of the way to get all primitives size is like following;

```c++
size_t all_primitives_size = 0;
for (const auto & route_section : route_msgs_.segments) {
  all_primitives_size += route_section.primitives.size();
}
route_lanelets_.reserve(all_primitives_size);

for (const auto & route_section : route_msgs.segments) {
...
}
```

This way can remove re-reserve but it makes the source code longer.
How do you think?

Reply:
Thank you!
Yes, indeed, this is better. Based on benchmark results, it seems that re-reserve might be slower.

I have changed this in #[143a489](https://github.com/autowarefoundation/autoware.universe/pull/1992/commits/143a4895d83daabc875f90895f53198535807563)